.github/workflows/ci.yml

```diff
@@ -10,8 +10,8 @@ on:
 
 env:
   # Common versions
-  GO_VERSION: '1.14'
-  GOLANGCI_VERSION: 'v1.31'
+  GO_VERSION: '1.16'
+  GOLANGCI_VERSION: 'v1.40.1'
```

Comment:
Is this change necessary? I generally like the idea of bumping the version of golangci-lint we're running, but unfortunately we actually run it multiple times during a build. One is triggered by `make lint` and uses the version [in the build submodule](https://github.com/upbound/build/blob/master/makelib/golang.mk#L99), which is v1.31. The other uses this GitHub Action. We run both because we like that the GitHub Action leaves comments on PRs with linter violations (and because I haven't got around to disabling `make lint` being called as part of the other CI jobs).

Reply:
This was the only way to make the CI happy. :thinking: Do you prefer to downgrade the linter?

Reply:
Ideally we'd bump to Go 1.16 but keep using golangci-lint 1.31. Can you provide some context on how that combination breaks CI?

Reply:
Yep, reverted that line, see the breaking CI: https://github.com/crossplane/crossplane-runtime/pull/267/checks?check_run_id=2817289334#step:7:74

Reply:
Anything we can do about this? (bumping the version fixes the issue)

Reply:
Reverted my change to fixup the CI again.

Reply:
@saschagrunert I believe I had similar issues with CI in [my PR](https://github.com/crossplane-contrib/provider-helm/pull/94), and able to work around with: https://github.com/crossplane-contrib/provider-helm/pull/94/commits/90bb4b3d012774134d1b02fa2ad1e6ff133434b4

Reply:
Alright, fixed as suggested and the CI seems green now.